build-system/tasks/runtime-test/helpers.js

```diff
@@ -0,0 +1,216 @@
+
+/**
+ * Extracts a mapping from CSS files to JS files from a well known file
+ * generated during `gulp css`.
+ *
+ * @return {!Object<string, string>}
+ */
+function extractCssJsFileMap() {
+  //TODO(esth): consolidate arg validation logic
+  if (!fs.existsSync(extensionsCssMapPath)) {
+    log(red('ERROR:'), 'Could not find the file',
+        cyan(extensionsCssMapPath) + '.');
+    log('Make sure', cyan('gulp css'), 'was run prior to this.');
+    process.exit();
+  }
+
+  const extensionsCssMap = fs.readFileSync(extensionsCssMapPath, 'utf8');
+  const extensionsCssMapJson = JSON.parse(extensionsCssMap);
+  const extensions = Object.keys(extensionsCssMapJson);
+  const cssJsFileMap = {};
+
+  // Adds an entry that maps a CSS file to a JS file
+  function addCssJsEntry(cssData, cssBinaryName, cssJsFileMap) {
+    const cssFilePath = `extensions/${cssData['name']}/${cssData['version']}/` +
+        `${cssBinaryName}.css`;
+    const jsFilePath = `build/${cssBinaryName}-${cssData['version']}.css.js`;
+    cssJsFileMap[cssFilePath] = jsFilePath;
+  }
+
+  extensions.forEach(extension => {
+    const cssData = extensionsCssMapJson[extension];
+    if (cssData['hasCss']) {
+      addCssJsEntry(cssData, cssData['name'], cssJsFileMap);
+      if (cssData.hasOwnProperty('cssBinaries')) {
+        const cssBinaries = cssData['cssBinaries'];
+        cssBinaries.forEach(cssBinary => {
+          addCssJsEntry(cssData, cssBinary, cssJsFileMap);
+        });
+      }
+    }
+  });
+  return cssJsFileMap;
+}
+
+/**
+ * Returns an array of ad types.
+ * @return {!Array<string>}
+ */
+function getAdTypes() {
+  const namingExceptions = {
+    // We recommend 3P ad networks use the same string for filename and type.
+    // Write exceptions here in alphabetic order.
+    // filename: [type1, type2, ... ]
+    adblade: ['adblade', 'industrybrains'],
+    mantis: ['mantis-display', 'mantis-recommend'],
+    weborama: ['weborama-display'],
+  };
+
+  // Start with Google ad types
+  const adTypes = ['adsense'];
+
+  // Add all other ad types
+  const files = fs.readdirSync('./ads/');
+  for (let i = 0; i < files.length; i++) {
+    if (path.extname(files[i]) == '.js'
+        && files[i][0] != '_' && files[i] != 'ads.extern.js') {
+      const adType = path.basename(files[i], '.js');
+      const expanded = namingExceptions[adType];
+      if (expanded) {
+        for (let j = 0; j < expanded.length; j++) {
+          adTypes.push(expanded[j]);
+        }
+      } else {
+        adTypes.push(adType);
+      }
+    }
+  }
+  return adTypes;
+}
+
+/**
+ * Returns the list of files imported by a JS file
+ *
+ * @param {string} jsFile
+ * @return {!Array<string>}
+ */
+function getImports(jsFile) {
+  const imports = findImports([jsFile], {
+    flatten: true,
+    packageImports: false,
+    absoluteImports: true,
+    relativeImports: true,
+  });
+  const files = [];
+  const rootDir = path.dirname(path.dirname(__dirname));
```

Comment:
Really late realization: `gulp test --local-changes` has been broken since this PR was checked in because this file (and therefore, `__dirname`) has moved a directory deeper, and therefore, `rootDir` needs an extra `path.dirname`. I'll send out a fix.

:)

Reply:
Okay sending out a fix in a bit!